shaptools/netweaver.py

```diff
@@ -147,6 +151,32 @@ def is_installed(self, sap_instance=None):
         else:
             raise ValueError('provided sap instance type is not valid: {}'.format(sap_instance))
 
+    @classmethod
+    def update_conf_file(cls, conf_file, **kwargs):
+        """
+        Update NW installation config file parameters. Add the parameters if they don't exist
+
+        Args:
+            conf_file (str): Path to the netweaver installation configuration file
+            kwargs (opt): Dictionary with the values to be updated.
+                Use the exact name of the netweaver configuration file
+
+        kwargs can be used in the next two modes:
+            update_conf_file(conf_file, sid='HA1', hostname='hacert01')
+            update_conf_file(conf_file, **{'sid': 'HA1', 'hostname': 'hacert01'})
+        """
+        for key, value in kwargs.items():
+            pattern = '{key}( *)=.*'.format(key=key)
+            new_value = '{key} = {value}'.format(key=key, value=value)
+            with open(conf_file, 'r+') as file_cache:
```

Comment:
I don't think we need to do this 2nd extra loop.
Why just not use a variable to store if pattern was found or not (check `if line == new_value`). And if this value is still false at the end of the 1st loop, just add the new value.

Reply:
Hi @arbulu89 the second for loop is for the case, if the pattern is found, and needed to be 'updated' for every occurrence.

Reply:
Yes, sorry... You are right... I confused the `open` with a `for`...